### PR TITLE
add client deadline to bridge interceptor

### PR DESCRIPTION
**Pull request by ivansenic**
**What this PR does**:
Adds client deadlines for each RPC call to the bridge. Configurable via the property.

**Which issue(s) this PR fixes**:
Internal.

### CODE REVIEW

**Comment by ivansenic** [comment]
@kathirsvn this is the Quarkus issue about the stub deadline: https://github.com/quarkusio/quarkus/issues/30082

File: apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/config/GrpcConfig.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import io.smallrye.config.WithDefault;
+import java.time.Duration;
+import java.util.Optional;
+
+/** Configuration for the gRPC calls to the Bridge. */
+@ConfigMapping(prefix = "stargate.grpc")
+public interface GrpcConfig {
+
+  /** @return Optional deadline duration for the each RPC to the bridge. Defaults to 30 seconds. */
+  @WithDefault("PT30S")
```
Comment (tatu-at-datastax) [inline apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/config/GrpcConfig.java:30]: Isn't 30 seconds kind of high?
I guess it is lower than what the ultimate default would be (or maybe there was no deadline specified at all).

Would think that something like 10 seconds might be better but maybe we might have some data to suggest something less than 30 seconds that is rarely seen.

**Review by tatu-at-datastax — APPROVED** [review]
Have one question on default value for grpc call deadline, but no strong objections to just merging. Makes sense to me & great to have a test to verify.

File: apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/config/GrpcConfig.java (hunk 1 above)
Comment (ivansenic) [reply apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/config/GrpcConfig.java:30]: * 30 seconds is what we agreed to be the value for now
* is it too much, maybe, but
   * better that not having at all (yes it's unlimited)
   * or to get timeout from the sidecar (60 sec default on nginx)
* and value is configurable, so you can set whatever you want when starting

Comment (tatu-at-datastax) [reply apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/config/GrpcConfig.java:30]: Ok.

**Pull request merged**